### PR TITLE
xds: fix potential access to missing control_plane field

### LINKED ISSUE #20458: xds: crash in onDiscoveryResponse
We experienced a crash in the XDS code path in our CI. Unfortunately we don't have a lot of debug info here (this is one of the few areas without core dumping support). Attached below are the logs

Envoy log: https://storage.googleapis.com/istio-prow/logs/integ-security_istio_postsubmit/1505969504671240192/artifacts/security-90e03050bd514c82bd47a1/TestReachability/_test_context/test-ns1-1-99926-state1182920515/primary-0/vm-v1-55b6b4d745-nnkx6_istio-proxy.envoy.err.log

Core dump: unfortunately not available

Config dump: we don't have the exact config, but it should be extremely similar to https://storage.googleapis.com/istio-prow/logs/integ-security_istio_postsubmit/1505969504671240192/artifacts/security-90e03050bd514c82bd47a1/TestReachability/_test_context/test-ns1-1-99926-state1182920515/primary-0/a-v1-7f7dc896d6-qzzj7_proxy-config.json which was another non-crashing workload running at the same time.

This doesn't seem like much to go on unfortunately

### PR DESCRIPTION
**Pull request by lambdai**
Commit Message:
Fix a potential control_plane access when the field doesn't exist.

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA] might fix #20458
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by lambdai** [comment]
the asan failure is likely just slow

```
[ RUN      ] IpVersions/ApiListenerIntegrationTest.Basic/IPv4
test/integration/api_listener_integration_test.cc:112: Failure
Value of: done.WaitForNotificationWithTimeout(absl::Seconds(1))
  Actual: false
Expected: true
Stack trace:
  0xfecc991: Envoy::(anonymous namespace)::ApiListenerIntegrationTest_Basic_Test::TestBody()
  0xfeccce0: Envoy::(anonymous namespace)::ApiListenerIntegrationTest_Basic_Test::TestBody()
  0x1a3eda78: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x1a3c12a3: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x1a38da3d: testing::Test::Run()
  0x1a38f171: testing::TestInfo::Run()
... Google Test internal frames ...

```

**Comment by daixiang0** [comment]
Do we need to cover those cases in tests? Original tests seems ignore them.

**Comment by lambdai** [comment]
This PR is a trivial fix to avoid invalid memory access. I don't plan to explore the control_plane propagation in this quick fix. There might be further clean up showing up while addressing https://github.com/envoyproxy/envoy/issues/20458

**Comment by daixiang0** [comment]
Sounds great.

**Review by adisuissa — APPROVED** [review]
Thanks for the fix, LGTM!

**Comment by adisuissa** [comment]
/assign @htuch 
For config-plane senior-maintainer review.

File: docs/root/version_history/current.rst
Comment (htuch) [inline docs/root/version_history/current.rst:61]: This is too vague, please rephrase.

**Review by htuch — COMMENTED** [review]
LGTM modulo version history nit.

**Comment by adisuissa** [comment]
Waiting for [comment](https://github.com/envoyproxy/envoy/pull/20465#discussion_r837037369) to be addressed.
/wait

File: docs/root/version_history/current.rst
Comment (lambdai) [reply docs/root/version_history/current.rst]: Update to
```
fix a crash that occurs when Envoy receives a discovery response without ``control_plane`` field.
```

SG?

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by lambdai** [comment]
check linux_x64 coverage seems stuck. Can anyone help or just merge the PR?

**Review by adisuissa — APPROVED** [review]

**Comment by htuch** [comment]
@lambdai coverage failure seems legit.

**Comment by adisuissa** [comment]
Can you please merge main? (should fix coverage)

**Comment by lambdai** [comment]
@adisuissa Merged. Thank you for the update!

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
